src/main/java/net/snowflake/ingest/connection/ServiceResponseHandler.java

```diff
@@ -76,7 +77,12 @@ public static IngestResponse unmarshallIngestResponse(HttpResponse response)
     }
 
     // grab the response entity
-    String blob = EntityUtils.toString(response.getEntity());
+    HttpEntity responseEntity = response.getEntity();
```

Comment:
Looks like all the logic are duplicate, can we do better?

Reply:
Also, can we add this to all APIs?

Reply:
I dont think there is any benefit of doing it in a separate method. 

I am doing this only for 3 snowpipe APIs for which confluent reported concerns. Hence I am pushing this only on top of `0.10.3`

I also plan to do it for all APIs though

Reply:
ok, given that this will be for all APIs, I think it makes sense to add a helper function to avoid duplication?

Reply:
Sounds good!